web/components/inputs/FcButton.vue

```diff
@@ -59,7 +69,31 @@ export default {
 
       const event = this.$analytics.buttonEvent(ihtml, this.$el);
       this.$analytics.send([event]);
+      return this.actionClickAuth();
+    },
+    actionClickAuth() {
+      const { auth: { loggedIn, user }, scope } = this;
+      if (scope === null) {
+        return true;
+      }
+      if (loggedIn) {
+        if (hasAuthScope(user, scope)) {
+          return true;
+        }
+        this.setDialog({
+          dialog: 'ConfirmUnauthorized',
+          dialogData: { scope },
+        });
+        return false;
```

Comment:
I wish this would cancel other event handlers, but sadly that appears to not be the case - so for buttons that visibly change the interface (such as our export / cancel toggles in Aggregate View), the user will see a brief flash of the new state before being redirected to login.